### PR TITLE
universal in config

### PR DESCRIPTION
**Pull request by tmsdkeys**
Closes: #41
Closes: #42

Right now, a lot of the just commands take a flag for universal channel or not.

This could be avoided if adding a flag to the config and separating out config files for different sets of contracts in one repo.

Expected outcomes:

will make commands even more easy
will reduce user input error
should provide automated check to see if contracts set match with the isUniversal flag

And...

The switch-client.js script now flips the config values first before updating the vibc addresses (dispatcher or universal channel handler). So it does reverse flipping the config when an error occurs on the update scripts

This can lead to errors requiring the user to update manually and/or do multiple switches

## Summary by CodeRabbit

- **New Features**
	- Introduced Custom IBC channels for advanced packet sending.
	- Added support for Universal channels, enhancing cross-network communication.
	- Implemented new scripts for dynamic contract deployment and configuration updates.
- **Enhancements**
	- Enhanced configuration management with new helper functions and scripts.
	- Improved deployment instructions and added detailed usage examples.
	- Streamlined packet sending with optimized scripts and configuration handling.
- **Documentation**
	- Updated the README with troubleshooting tips, dependency checks, and detailed instructions for new features.
- **Refactor**
	- Refactored scripts for better readability, configuration loading, and error handling.
	- Updated script names and paths for clarity and consistency.
- **Bug Fixes**
	- Fixed issues in channel creation and packet sending by adjusting timing and event listening logic.

### CODE REVIEW

File: scripts/_sanity-check.js [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
 function runSanityCheck(network) {
     // Example: 
-    // $ node scripts/sanity-check.js false
-    const isUniversalChannel = process.argv[2].toLowerCase();
+    // $ node scripts/sanity-check.js
+    const configPath = getConfigPath();
+    const config = require(configPath);
 
-    const scriptSuffix = isUniversalChannel === 'true' ? 'universal' : 'custom';
+    const scriptSuffix = config.isUniversal ? 'universal' : 'custom';
```
[the remark this answers is not in the document]
Comment (tmsdkeys) [reply scripts/_sanity-check.js:10]: What do you suggest for sanitization?

File: figcon.json
Comment (kenobijon) [inline figcon.json:1]: What is the purpose of this ? Is this referenced anywhere else?

File: scripts/_create-channel-config.js
Comment (kenobijon) [inline scripts/_create-channel-config.js:1]: getting the following error 
```
ReferenceError: network is not defined
```

**Review by kenobijon — CHANGES_REQUESTED** [review]
fix the network check in `createChannelAndCapture` then looks good

File: figcon.json
Comment (tmsdkeys) [reply figcon.json:1]: Whoops, I forgot to do `git restore --staged config.json figcon.json` here...

Will do in subsequent commit

File: scripts/_create-channel-config.js
Comment (tmsdkeys) [reply scripts/_create-channel-config.js:1]: Checking

File: README.md [hunk 2]
```diff
@@ -89,17 +128,12 @@ It makes sure you've got the correct contracts set, deploys new instances, creat
 > Note: by default the sim-client is used to improve latency. This is useful for iterative development and testing BUT also insecure as it involves no proofs. Make sure to move to the client **with proofs** by running another just command...
 
 ```bash
-# Usage: just switch-client [universal=true]
+# Usage: just switch-client
 just switch-client false
```
Comment (kenobijon) [inline README.md:132]: switch-client no longer has a true/false option

**Pull request merged**